openmetadata-ui/src/main/resources/ui/src/components/TeamsSelectable/TeamsSelectable.tsx

```diff
@@ -29,22 +28,30 @@ interface CustomOption extends SelectableOption {
 
 interface Props {
   onSelectionChange: (teams: string[]) => void;
+  filterJoinable?: boolean;
+  placeholder?: string;
 }
 
-const TeamsSelectable = ({ onSelectionChange }: Props) => {
+const TeamsSelectable = ({
+  onSelectionChange,
+  filterJoinable,
+  placeholder = 'Type to search for teams',
+}: Props) => {
   const [teamSearchText, setTeamSearchText] = useState<string>('');
 
   const handleSelectionChange = (selectedOptions: SelectableOption[]) => {
     onSelectionChange(selectedOptions.map((option) => option.value));
   };
 
   const getOptions = (teams: Team[]) => {
-    return teams
-      .filter((team) => team.isJoinable)
-      .map((team) => ({
-        label: getEntityName(team as EntityReference),
-        value: team.id,
-      }));
+    const filteredTeams = filterJoinable
+      ? teams.filter((team) => team.isJoinable)
```

Comment:
We should avoid client-side filtering and use filtering with the query. Because this can potentially return 0 results if the first n results returned do not match the filter criteria.

Reply:
For suggest API we don't have BE filtering supported and for query API I have already added BE Filter please check